### PR TITLE
Indicator Component

### PR DESCRIPTION
**Pull request by xingruoxicic**
Screen Shot 2023-03-15 at 11 49 52 AM

### CODE REVIEW

File: core-library/component-styling/indicator/_indicator.scss [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    }
+  }
+
+  .number,
+  .dot {
+    border-radius: 50%;
+  }
+
+  .rounded {
+    border-radius: 50% !important;
```
Comment (MikeJalow) [inline core-library/component-styling/indicator/_indicator.scss:201]: is there any way we could avoid using !important? We'd ideally never use this in our DS.

File: core-library/component-styling/indicator/_indicator.scss [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+  }
+
+  &.critical,
+  &.red {
+    color: var(--critical-text);
+  }
+
+  &.generic,
+  &.grey {
+    color: color.mix-token(neutral, 900);
```
Comment (MikeJalow) [inline core-library/component-styling/indicator/_indicator.scss:117]: these mix-tokens will need to moved to a fresh token sheet - if we don't it'll cause issues when we're trying to integrate dark mode.

**Review by MikeJalow — CHANGES_REQUESTED** [review]

File: component-library/component-lib/src/lib/shared/indicator/indicator.component.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  @Input() config: IIndicatorConfig = {
+    type: 'text',
+    category: IndicatorTreatment.weak,
+    purpose: IndicatorPurpose.status
+  };
+  @ViewChild('label') label?: ElementRef<HTMLDivElement>;
+  EIndicatorStatus = IndicatorStatus;
+  rounded?: boolean;
+  abbr?: boolean; // Display abbr tag when text is truncated
+  constructor() { }
```
Comment (kristopher-houston) [inline component-library/component-lib/src/lib/shared/indicator/indicator.component.ts:73]: Remove if not used

**Review by kristopher-houston — APPROVED** [review]

File: component-library/component-lib/src/lib/shared/indicator/indicator.component.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    type: 'text',
+    category: IndicatorTreatment.weak,
+    purpose: IndicatorPurpose.status
+  };
+  @ViewChild('label') label?: ElementRef<HTMLDivElement>;
+  EIndicatorStatus = IndicatorStatus;
+  rounded?: boolean;
+  abbr?: boolean; // Display abbr tag when text is truncated
+
+  ngOnInit(): void {
```
Comment (kristopher-houston) [inline component-library/component-lib/src/lib/shared/indicator/indicator.component.ts:74]: Remove ': void'

File: component-library/component-lib/src/lib/shared/indicator/indicator.component.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    this.checkLabelRounded();
+    this.checkNumber();
+  }
+
+  ngAfterViewInit() {
+    this.checkLabelLength();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
```
Comment (kristopher-houston) [inline component-library/component-lib/src/lib/shared/indicator/indicator.component.ts:83]: Remove ': void'

**Review by MikeJalow — APPROVED** [review]

**Pull request merged**